packages/medusa/src/services/customer-group.ts

```diff
@@ -86,6 +89,36 @@ class CustomerGroupService extends BaseService {
     })
   }
 
+  /**
+   * Update a customer group.
+   *
+   * @param {string} customerGroupId - id of the customer group
+   * @param {CustomerGroupUpdate} update - customer group partial data
+   */
+  async update(
+    customerGroupId: string,
+    update: CustomerGroupUpdate
+  ): Promise<CustomerGroup[]> {
+    return this.atomicPhase_(async (manager) => {
+      const { name, metadata } = update
+
+      const cgRepo: CustomerGroupRepository = manager.getCustomRepository(
+        this.customerGroupRepository_
+      )
+
+      const customerGroup = await this.retrieve(customerGroupId)
+
+      if (name) {
+        customerGroup.name = name
+      }
+
+      if (typeof metadata !== "undefined") {
+        customerGroup.metadata = metadata
+      }
```

Comment:
Only a suggestion:

In other services, we've used the following approach for the update logic to ensure, that if we add additional fields to the `CustomerGroupUpdate` in the future, we don't have to add a new check for every additional prop:
```
const { metadata, ...rest } = update

for (const key of Object.keys(rest).filter((k) => rest[k] !== undefined)) {
        customerGroup[key] = rest[key]
}

if (typeof metadata !== "undefined") {
        customerGroup.metadata = metadata
}
```

Let me know, what you think? It's prob. a case of over engineering it at this point but thought I'd throw the comment in here anyways 👍

Reply:
I agree -- easy future proof gains 😁